careers/ladder.md

```diff
@@ -563,6 +578,65 @@ See
 [this document](https://www.notion.so/artsy/Artsy-Tech-Lead-Expanded-Responsibilities-Public-710ee0cd2fce47dcaa722ece80e365b2)
 for an expanded set of Tech Lead Responsibilities at Artsy.
 
+## Staff Engineer Archetypes
```

Comment:
Related to Devon's comment, let's add a section that links out to some of the sources we used for inspiration. 😄

Reply:
Might also be worth an introductory sentence here that explains the purpose of this section.

There we can call out things like:
- Getting to the staff level requires having an engineering org-wide impact, but there are many ways to do that.
- These archetypes are not prescriptions, but intended to serve as a tool to help individuals lean into their strengths and business needs in order to reach the staff level.

(something like that...)

Reply:
I've added these to the TODO list in the PR description, and will work on them tomorrow.